modules/potree/src/lib/potree-node-source.ts

```diff
@@ -0,0 +1,119 @@
+import {load} from '@loaders.gl/core';
+import {MeshGeometry} from '@loaders.gl/schema';
+import {DataSource, DataSourceProps, LoaderOptions, resolvePath} from '@loaders.gl/loader-utils';
+import {LASLoader} from '@loaders.gl/las';
+import {PotreeMetadata} from '../types/potree-metadata';
+import {POTreeNode} from '../parsers/parse-potree-hierarchy-chunk';
+import {PotreeHierarchyChunkLoader} from '../potree-hierarchy-chunk-loader';
+import {PotreeLoader} from '../potree-loader';
+
+export type PotreeNodesSourceProps = DataSourceProps & {
+  attributions?: string[];
+  potree?: {
+    loadOptions?: LoaderOptions; // PotreeLoaderOptions;
+    // TODO - add options here
+  };
+};
+
+/**
+ * A Potree data source
+ * @note Point cloud nodes tile source
+ */
+export class PotreeNodesSource extends DataSource {
+  url: string;
+  data: string | Blob;
+  props: PotreeNodesSourceProps;
+  metadata: PotreeMetadata | null = null;
+  root: POTreeNode | null = null;
+  isReady = false;
+
+  private initPromise: Promise<void> | null = null;
+
+  get isSupported(): boolean {
+    return (
+      this.isReady &&
+      ['1.4', '1.7'].includes(this.metadata?.version ?? '') &&
+      typeof this.metadata?.pointAttributes === 'string' &&
+      ['LAS', 'LAZ'].includes(this.metadata?.pointAttributes)
+    );
+  }
+
+  get contentExtension(): string | null {
+    if (!this.isReady) {
+      return null;
+    }
+    switch (this.metadata?.pointAttributes) {
+      case 'LAS':
+        return 'las';
+      case 'LAZ':
+        return 'laz';
+      default:
+        return 'bin';
+    }
+  }
+
+  constructor(data: string | Blob, props: PotreeNodesSourceProps) {
+    super(props);
+    this.props = props;
+    this.data = data;
+    this.url = typeof data === 'string' ? resolvePath(data) : '';
+
+    this.initPromise = this.init();
+  }
+
+  async init() {
+    this.metadata = await load(`${this.url}/cloud.js`, PotreeLoader);
+    await this.loadHierarchy();
+    this.isReady = true;
+  }
+
+  async loadNodeContent(path: number[]): Promise<MeshGeometry | null> {
+    await this.initPromise;
+
+    if (!this.isSupported) {
+      return null;
+    }
+
+    const isAvailable = await this.getNodeAvailability(path);
+    if (isAvailable) {
+      return load(
+        `${this.url}/${this.metadata?.octreeDir}/r/r${path.join()}.${this.contentExtension}`,
+        LASLoader
+      );
+    }
+    return null;
+  }
+
+  async loadHierarchy(): Promise<void> {
+    await this.initPromise;
+    this.root = await load(
+      `${this.url}/${this.metadata?.octreeDir}/r/r.hrc`,
+      PotreeHierarchyChunkLoader
+    );
+  }
+
+  async getNodeAvailability(path: number[]): Promise<boolean> {
```

Comment:
Nit: Playing with the naming... not sure what feels best.
```suggestion
  async isNodeAvailable(path: number[]): Promise<boolean> {
```

Reply:
Done